### PR TITLE
Product listing

### PR DESCRIPTION
**Pull request by Meghana-sk**
- Added product details and assets to database
- Fetch data and display on product listing page
- **_File changes are more as more assets are added_**
- Filter functionality yet to be implemented

### CODE REVIEW

File: src/pages/Home/Home.jsx [hunk 1]
```diff
@@ -1,7 +1,22 @@
...
+        try {
+            const productsData = await axios.get("/api/categories");
+            setProductCategories(productsData.data.categories);
+        } catch (error) {
+            console.error(error)
+        }
+    }
+
+    useEffect(() => {
+        getProductCategories()
```
Comment (dipteekhd) [inline src/pages/Home/Home.jsx]: can write IIFE here instead of creating getProductCategories function separately(Suggestion)

Comment (Meghana-sk) [reply src/pages/Home/Home.jsx]: Sure.

File: src/index.js [hunk 2]
```diff
@@ -4,15 +4,18 @@ import "./index.css";
...
-    <App />
-  </BrowserRouter>
-  </React.StrictMode>,
-  document.getElementById("root")
+	<React.StrictMode>
+    	<BrowserRouter>
+    		<ProductsProvider>
+				<App />	
+    		</ProductsProvider>
+  		</BrowserRouter>
```
Comment (Kevin-Solomon) [inline src/index.js:18]: Use Prettier for better indentation.

File: src/pages/Home/Home.jsx [hunk 3]
```diff
@@ -1,7 +1,22 @@
...
+import { useState, useEffect } from "react";
 
 const Home = () => {
+    const [ productCategories, setProductCategories] = useState([]);
+    const getProductCategories = async() => {
+        try {
+            const productsData = await axios.get("/api/categories");
+            setProductCategories(productsData.data.categories);
+        } catch (error) {
+            console.error(error)
```
Comment (Kevin-Solomon) [inline src/pages/Home/Home.jsx]: You can check for the response status here.
Just to be extra safe.

**Review by Kevin-Solomon — APPROVED** [review]

File: src/pages/Products/Products.jsx [hunk 4]
```diff
@@ -1,22 +1,28 @@
...
+                        {
+                            availableProducts.map(({
+                                _id,
+                                title,
+                                price, 
+                                subtitle, 
+                                rating, 
+                                imgSrc
+                                }) => (
+                                <Card key={_id} price={price} title={title} subtitle={subtitle} rating={rating} imgSrc={imgSrc}/>
```
Comment (dipteekhd) [inline src/pages/Products/Products.jsx]: can Pass **product** as prop to Card Component and destructure it in Card Component for easy syntax (Suggestion)

**Review by dipteekhd — APPROVED** [review]

**Pull request merged**